plugins/inputs/teamspeak/teamspeak.go

```diff
@@ -0,0 +1,90 @@
+package teamspeak
+
+import (
+	"github.com/multiplay/go-ts3"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"strconv"
+)
+
+type Teamspeak struct {
+	Server   string
+	Username string
+	Password string
+	Vservers []int
+}
+
+func (ts *Teamspeak) Description() string {
+	return "Reads metrics from a Teamspeak 3 Server via ServerQuery"
+}
+
+const sampleConfig = `
+  ## Server address for Teamspeak 3 ServerQuery
+  # server = "127.0.0.1:10011"
+  ## Username for ServerQuery
+  # username = "serveradmin"
+  ## Password for ServerQuery
+  # password = "secret"
+  ## Array of virtual servers
+  # vservers = [1]
+`
+
+func (ts *Teamspeak) SampleConfig() string {
+	return sampleConfig
+}
+
+func (ts *Teamspeak) Gather(acc telegraf.Accumulator) error {
+	var err error
+	client, err := ts3.NewClient(ts.Server)
```

Comment:
Maybe we should make the client once and also possibly login once?  This might be a good or bad idea depending on how the teamspeak client handles disconnects.

Reply:
Thanks for the review! I think thats a good idea. It also reduces logging very strongly on the server side. I changed it with my latest commit.